R/client-cloud.R

```diff
@@ -230,12 +234,21 @@ cloudClient <- function(service, authInfo) {
         revision$application_id
     },
 
-    deployApplication = function(application, bundleId = NULL) {
+    deployApplication = function(application, bundleId = NULL, spaceId = NULL) {
+      outputPatchData <- list()
+
       currentProjectId <- getCurrentProjectId(service, authInfo)
       if (!is.null(currentProjectId)) {
+        outputPatchData$project <- currentProjectId
```

Comment:
This is different logic than is used when creating; intentional? In this case, we could end up using `currentProjectId` from the environment and the `spaceId` argument. When creating, the `spaceId` argument is only used when there is no `currentProjectId`.

Reply:
I think the logic as-written makes sense (at least to me). When you deploy, it should definitely set the space based on the space of the environment project. But on redeploys, it would probably be more confusing for projects to move between spaces without providing the argument.

@m-- Any other thoughts?

Reply:
I think that makes sense to me.